packages/flutter/test/material/carousel_test.dart

```diff
@@ -0,0 +1,461 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Carousel defaults', (WidgetTester tester) async {
+    final ThemeData theme = ThemeData();
+    final ColorScheme colorScheme = theme.colorScheme;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: theme,
+        home: Scaffold(
+          body: Carousel(
+            itemExtent: 200,
+            children: List<Widget>.generate(10, (int index) {
+              return Center(child: Text('Item $index'));
+            }),
+          ),
+        ),
+      ),
+    );
+
+    final Finder carouselMaterial = find.descendant(
+      of: find.byType(Carousel),
+      matching: find.byType(Material),
+    ).first;
+
+    final Material material = tester.widget<Material>(carouselMaterial);
+    expect(material.clipBehavior, Clip.antiAlias);
+    expect(material.color, colorScheme.surface);
+    expect(material.elevation, 0.0);
+    expect(material.shape, const RoundedRectangleBorder(
+      borderRadius: BorderRadius.all(Radius.circular(28.0))
+    ));
+  });
+
+  testWidgets('Carousel items customization', (WidgetTester tester) async {
+    final GlobalKey key = GlobalKey();
```

Comment:
```suggestion
    final Key key = UniqueKey();
```

(I don't think you actually need a global key here)